packages/@webex/internal-plugin-calendar/src/calendar.js

```diff
@@ -312,6 +349,118 @@ const Calendar = WebexPlugin.extend({
+   * Update calendar event
+   * @param {string} [id] calendar event id
+   * @param {object} [data] meeting payload data
+   * @param {object} [query] the query parameters for specific usage
+   * @returns {Promise} Resolves with updating calendar event response
+   * */
+  updateCalendarEvent(id, data, query) {
+    return EncryptHelper.encryptCalendarEventRequest(this, data).then(() => {
+      return this.request({
+        method: 'PATCH',
+        service: 'calendar',
+        body: data,
+        resource: `calendarEvents/${base64.encode(id)}/sync`,
+        qs: query || {},
+      });
+    });
+  },
+
+  /**
+   * Delete calendar event
+   * @param {string} [id] calendar event id
+   * @returns {Promise} Resolves with deleting calendar event response
+   * */
+  deleteCalendarEvent(id) {
+    return this.request({
+      method: 'DELETE',
+      service: 'calendar',
+      resource: `calendarEvents/${base64.encode(id)}/sync`,
+    });
+  },
+
+  /**
+   * @typedef QuerySchedulerDataOptions
+   * @param {string} [siteName] it is site full url, must have. Example: ccctest.dmz.webex.com
+   * @param {string} [id] it is seriesOrOccurrenceId. If present, the series/occurrence meeting ID to fetch data for. It should be base64 encoded.
+   *                      Example: 040000008200E00074C5B7101A82E008000000004A99F11A0841D9010000000000000000100000009EE499D4A71C1A46B51494C70EC7BFE5
+   * @param {string} [clientMeetingId] If present, the client meeting UUID to fetch data for. It should be base64 encoded.
+   *                      Example: 7f318aa9-887c-6e94-802a-8dc8e6eb1a0a
+   * @param {string} [scheduleTemplateId] it template id.
+   * @param {string} [sessionTypeId] it session type id.
+   * @param {string} [organizerCIUserId] required in schedule-on-behalf case. It is the organizer's CI UUID.
+   * @param {boolean} [usmPreference]
+   * @param {string} [webexMeetingId] webex side meeting UUID
+   * @param {string} [eventId] event ID.
+   * @param {string} [icalUid] icalendar UUID.
+   * @param {string} [thirdPartyType] third part type, such as: Microsoft
+   */
+  /**
+   * Get scheduler data from calendar service
+   * @param {QuerySchedulerDataOptions} [query] the command parameters for fetching scheduler data.
+   * @returns {Promise} Resolves with a decrypted scheduler data
+   * */
+  getSchedulerData(query) {
+    return this.request({
+      method: 'GET',
+      service: 'calendar',
+      resource: 'schedulerData',
+      qs: query || {},
+    }).then((response) => {
+      return DecryptHelper.decryptSchedulerDataResponse(this, response.body).then(() => {
+        delete response.body.encryptionKeyUrl;
+
+        return response;
+      });
+    });
+  },
+
+  /**
+   * Get free busy status from calendar service
+   * @param {Object} [data] the command parameters for fetching free busy status.
+   * @returns {Promise} Resolves with a decrypted response
+   * */
+  getFreeBusy(data) {
+    return new Promise((resolve, reject) => {
```

Comment:
same here await

Reply:
This is an interesting use of deferred promises, is this necessary here? Also, these promises seem to resolve pre-maturely without returning promises in the `then` clauses.

Reply:
Yes, it is because the free busy API is an async API call, this API does not return real free busy data immediately.

Once the free busy data is collected and ready to return, another mercury event will be triggered, we listen it and use '_handleFreeBusy' function to resolve deferred promise.

The caller want to use this API simple, and does not want to listen another event for returned data separately, so I have to implement it this way.

Do you have any better idea? thanks a lot!

Reply:
I thought this could be why it was deferred, we do this in a few other places [deferring a promise to an event]. This should be okay.